streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamPrintTest.java

```diff
@@ -41,84 +42,83 @@
                 return String.format("%d, %s", key, value);
             }
         };
-        final KStreamPrint<Integer, String> kStreamPrint = new KStreamPrint<>(new PrintForeachAction<>(printWriter, mapper, "test-stream"), intSerd, stringSerd);
+
+        kStreamPrint = new KStreamPrint<>(new PrintForeachAction<>(printWriter, mapper, "test-stream"), intSerd, stringSerd);
+    }
+
+    @Test
+    public void testPrintStreamWithProvidedKeyValueMapper() {
 
         final List<KeyValue<Integer, String>> inputRecords = Arrays.asList(
                 new KeyValue<>(0, "zero"),
                 new KeyValue<>(1, "one"),
                 new KeyValue<>(2, "two"),
                 new KeyValue<>(3, "three"));
-        
+
         final String[] expectedResult = {"[test-stream]: 0, zero", "[test-stream]: 1, one", "[test-stream]: 2, two", "[test-stream]: 3, three"};
-        
-        final StreamsBuilder builder = new StreamsBuilder();
-        final KStream<Integer, String> stream = builder.stream(intSerd, stringSerd, topicName);
-        stream.process(kStreamPrint);
-        
-        driver.setUp(builder);
+
+        Processor printProcessor = kStreamPrint.get();
+        ProcessorContext processorContext = EasyMock.createNiceMock(ProcessorContext.class);
+        EasyMock.replay(processorContext);
+
+        printProcessor.init(processorContext);
         for (KeyValue<Integer, String> record: inputRecords) {
-            driver.process(topicName, record.key, record.value);
+            printProcessor.process(record.key, record.value);
         }
         printWriter.flush();
-        final String[] flushOutDatas = new String(byteOutStream.toByteArray(), Charset.forName("UTF-8")).split("\\r*\\n");
-        for (int i = 0; i < flushOutDatas.length; i++) {
-            assertEquals(expectedResult[i], flushOutDatas[i]);
-        }
+        assertFlushData(expectedResult, byteOutStream);
     }
 
     @Test
-    public void testPrintStreamWithProvidedKeyValueMapper() {
-        final KeyValueMapper<Integer, String, String> mapper = new KeyValueMapper<Integer, String, String>() {
-            @Override
-            public String apply(Integer key, String value) {
-                return String.format("(%d, %s)", key, value);
-            }
-        };
-        final KStreamPrint<Integer, String> kStreamPrint = new KStreamPrint<>(new PrintForeachAction<>(printWriter, mapper, "test-stream"), intSerd, stringSerd);
+    public void testPrintKeyValueStringBytesArray() {
 
-        final List<KeyValue<Integer, String>> inputRecords = Arrays.asList(
-                new KeyValue<>(0, "zero"),
-                new KeyValue<>(1, "one"),
-                new KeyValue<>(2, "two"),
-                new KeyValue<>(3, "three"));
+        // we don't have a topic name because we don't need it for the test at this level
+        final List<KeyValue<Integer, byte[]>> inputRecords = Arrays.asList(
```

Comment:
We should make this test be fore `<byte[], byte[]>` as then we have covered both key and value deserialization.